libs/logging/src/export.ts

```diff
@@ -0,0 +1,122 @@
+import { assert, lines } from '@votingworks/basics';
+import {
+  Dictionary,
+  ElectionDefinition,
+  EventLogging,
+  safeParse,
+  safeParseJson,
+} from '@votingworks/types';
+import { JsonStreamInput, jsonStream } from '@votingworks/utils';
+import { z } from 'zod';
+import { LogEventId } from './log_event_ids';
+import { CLIENT_SIDE_LOG_SOURCES } from './log_source';
+import { type Logger } from './logger';
+import { DEVICE_TYPES_FOR_APP, LogLineSchema, LoggingUserRole } from './types';
+
+function extractAdditionalKeysFromObj(
+  innerObj: Dictionary<string>,
+  outerObj: Dictionary<string>
+): Dictionary<string> {
+  const baseDict: Dictionary<string> = {};
+  return Object.keys(outerObj)
+    .filter((key) => !(key in innerObj))
+    .reduce((res, nextKey) => {
+      res[nextKey] = outerObj[nextKey];
+      return res;
+    }, baseDict);
+}
+
+async function* generateCdfEventsForExport(
+  logger: Logger,
+  currentUser: LoggingUserRole,
+  logFileReader: AsyncIterable<string>
+): AsyncGenerator<EventLogging.Event> {
+  const logs = lines(logFileReader).filter((l) => l !== '');
```

Comment:
Aside from changing the function type signature, this was most of the actual functional change. Instead of `rawLogFileContents.split('\n').filter(…)` we have `lines(logFileReader).filter(…)`.